libraries/botbuilder-dialogs/tests/dialogHelper.test.js

```diff
@@ -131,18 +128,55 @@ class SimpleComponentDialog extends ComponentDialog {
     async promptForName(step) {
         return step.prompt(this.TextPrompt, {
             prompt: MessageFactory.text('Hello, what is your name?', undefined, InputHints.ExpectingInput),
-            retryPrompt: MessageFactory.text('Hello, what is your name again?', undefined, InputHints.ExpectingInput)
+            retryPrompt: MessageFactory.text('Hello, what is your name again?', undefined, InputHints.ExpectingInput),
         });
     }
 
     async finalStep(step) {
-        await step.context.sendActivity(`Hello ${ step.result }, nice to meet you!`);
+        await step.context.sendActivity(`Hello ${step.result}, nice to meet you!`);
         return step.endDialog(step.result);
     }
+}
+
+class ComponentDialogWithPrematureEnd extends ComponentDialog {
+    constructor() {
+        super('ComponentDialogWithPrematureEnd');
+        this.WaterfallDialog = 'WaterfallDialog';
+        this.addDialog(new WaterfallDialog(this.WaterfallDialog, [this.finalStep.bind(this)]));
+        this.initialDialogId = this.WaterfallDialog;
```

Comment:
Using a locally-scope variable for `'WaterfallDialog'` will suffice for registering the dialog and the `ComponentDialog.initialDialogId`. There are no references to `.WaterfallDialog` outside of the constructors in this file.